### PR TITLE
Use case: resize image

### PR DESCRIPTION
**Pull request by killercup**
[This](https://rust-lang-nursery.github.io/rust-cookbook/concurrency.html#generate-jpg-thumbnails-in-parallel) is a nice script I found in the cookbook that is, to me, a great use case for quicli.

It uses glob, rayon, and image. I think of these three, glob and rayon are general purpose for CLI tools but dealing with image is more special purpose.

I'm not entirely happy with the code as-is, and will comment on some things that I think could be improved. In general, I'm still really impressed by how great Rust is for stuff like this :)

### CODE REVIEW

File: examples/thumbs.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+//! From <https://rust-lang-nursery.github.io/rust-cookbook/concurrency.html#generate-jpg-thumbnails-in-parallel>
+
+extern crate image;
+#[macro_use]
+extern crate quicli;
+extern crate structopt;
+
+use quicli::prelude::*;
+
+use std::path::{Path, PathBuf};
```
Comment (killercup) [inline examples/thumbs.rs]: Maybe you are right (#20), @vitiral, and we should re-export some functions for dealing with paths.

File: examples/thumbs.rs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    /// Where do you want to save the thumbnails?
+    #[structopt(long = "output", short = "o", default_value = "thumbnails")]
+    thumb_dir: String,
+    /// Pass many times for more log output
+    #[structopt(long = "verbosity", short = "v")]
+    verbosity: u64,
+}
+
+main!(|args: Cli, log_level: verbosity| {
+    let files: Vec<_> = glob(&args.pattern)?.filter_map(|x| x.ok()).collect();
```
Comment (killercup) [inline examples/thumbs.rs]: Quoting the commit adding `glob`:

> I'm not really happy with [the `glob`] name.
> 
> Maybe we should directly convert this to a `Vec`? And skip all files we
> can't read? I think that currently, you'll write this a lot:
> 
> ```rust
> let files = glob("*.txt")?.filter_map(|x| x.ok()).collect();
> ```

File: examples/thumbs.rs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    create_dir(&args.thumb_dir)?;
+
+    info!("Saving {} thumbnails into {:?}...", files.len(), args.thumb_dir);
+
+    let image_failures: Vec<_> = files
+        .par_iter()
+        .map(|path| {
+            make_thumbnail(path, &args.thumb_dir, args.size).map_err(|e| {
```
Comment (killercup) [inline examples/thumbs.rs]: So, this is ugly. I want to:

- Print nice errors as soon as they happen
- count the errors later on (would require an AtomicI32 if done in a loop)

File: examples/thumbs.rs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+struct Cli {
+    /// Which files?
+    #[structopt(default_value = "*.jpg")]
+    pattern: String,
+    /// How long should the longest edge of the thumbnail be?
+    #[structopt(long = "max-size", short = "s", default_value = "300")]
+    size: u32,
+    /// Where do you want to save the thumbnails?
+    #[structopt(long = "output", short = "o", default_value = "thumbnails")]
+    thumb_dir: String,
```
Comment (killercup) [inline examples/thumbs.rs]: ~~Structopt can't be used for PathBuf~~ Actually just wanted to open an issue there and found <https://github.com/TeXitoi/structopt/issues/35>. `#[structopt(help = "Output filename", parse(from_os_str))]` is not very intuitive, though.

**Comment by killercup** [comment]
Let's merge this and release a 0.1.2 before the weekend is over 😅

**Comment by vitiral** [comment]
sorry I was so slow! Put up a PR for a couple of grammar errors I saw.

**Pull request merged**